### PR TITLE
Update CMake code and switch to CPM.

### PR DESCRIPTION
**Pull request by jdumas**
Notable changes:
- Removed `SUITE_SPARSE_WITH_MKL`. SuiteSparse always needs a BLAS implementation. Which one is picked is dictated by the `blas.cmake` and `lapack.cmake` content.
- Removed setting `EIGEN_WITH_MKL` automatically if `POLYSOLVE_WITH_MKL` is enabled. Whether we have the MKL solver available, and whether Eigen uses MKL for matrix ops should be two different settings.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -89,13 +105,10 @@ list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/polysolve/")
 list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/recipes/")
 list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/find/")
 
-# Color output
+# General CMake utils
+include(polysolve_cpm_cache)
 include(polysolve_use_colors)
 
-# IPC Toolkit utils
-include(polysolve_prepend_current_path)
```
Comment (zfergus) [inline CMakeLists.txt:96]: Good to be ride of these things 😁

File: CMakeLists.txt [hunk 2]
```diff
@@ -110,7 +123,7 @@ set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 add_library(polysolve)
 add_library(polysolve::polysolve ALIAS polysolve)
 
-add_subdirectory(src)
+add_subdirectory(src/polysolve)
```
Comment (zfergus) [inline CMakeLists.txt:126]: Glad you changed this.

**Comment by zfergus** [comment]
I am curious, why switch from FetchContent to CPM? I know CPM is a wrapper on top of FetchContent. What features make it worth using?

**Review by teseoch — APPROVED** [review]

**Comment by jdumas** [comment]
> I am curious: why switch from FetchContent to CPM? I know CPM is a wrapper on top of FetchContent. What features make it worth using?

Two things:
- When there is nothing to do, CPM is _much_, _much_ faster than plain FetchContent. So re-runnning CMake on your project is blazing fast.
- CPM has the ability to cache source outside of the build folder. With FetchContent, we can set the download path to be something outside the build folder, but it's not exactly the same. Using CPM to cache download of third-party files makes it much faster to recompile a project.

**Pull request merged**